lib/macho/macho_file.rb

```diff
@@ -409,16 +409,25 @@ def add_rpath(path, _options = {})
     #  file.delete_rpath("/lib")
     #  file.rpaths # => []
     # @param path [String] the runtime path to delete
-    # @param _options [Hash]
+    # @param options [Hash]
+    # @option options [Boolean] :uniq (false) if true, also delete
+    #  duplicates of the requested path. If false, delete the first
+    #  instance (by offset) of the requested path.
     # @return void
     # @raise [RpathUnknownError] if no such runtime path exists
-    # @note `_options` is currently unused and is provided for signature
-    #  compatibility with {MachO::FatFile#delete_rpath}
-    def delete_rpath(path, _options = {})
-      rpath_cmd = command(:LC_RPATH).find { |r| r.path.to_s == path }
-      raise RpathUnknownError, path unless rpath_cmd
+    def delete_rpath(path, options = {})
+      uniq = options.fetch(:uniq, false)
+      search_method = uniq ? :select : :find
+
+      rpath_cmds = command(:LC_RPATH).method(search_method).call { |r| r.path.to_s == path }
+      raise RpathUnknownError, path if rpath_cmds.nil? || rpath_cmds == []
 
-      delete_command(rpath_cmd)
+      if uniq
+        # delete the commands in reverse order, offset descending.
+        rpath_cmds.reverse_each { |cmd| delete_command(cmd) }
+      else
+        delete_command(rpath_cmds)
+      end
```

Comment:
Same comment as above.

Reply:
I wonder if this is idiomatic Ruby...
```suggestion
      # delete the commands in reverse order, offset descending.
      return rpath_cmds.reverse_each { |cmd| delete_command(cmd) } if uniq
      delete_command(rpath_cmds)
```